src/lib/tabs/tabCharacters/CharacterGrid.tsx

```diff
@@ -17,74 +12,95 @@ import {
 import i18next from 'i18next'
 import { Assets } from 'lib/rendering/assets'
 import DB from 'lib/state/db'
+import { CharacterTabController } from 'lib/tabs/tabCharacters/characterTabController'
+import { useCharacterTabStore } from 'lib/tabs/tabCharacters/useCharacterTabStore'
 import {
-  MutableRefObject,
+  useCallback,
   useMemo,
+  useRef,
   useState,
 } from 'react'
 import { useTranslation } from 'react-i18next'
 import { Character } from 'types/character'
 
 const { Text } = Typography
 
-// FIXME HIGH
-
-export function CharacterGrid(props: {
-  characterGrid: MutableRefObject<AgGridReact<Character>>,
-  cellClickedListener: (x: CellClickedEvent<Character>) => void,
-  cellDoubleClickedListener: (x: CellDoubleClickedEvent<Character>) => void,
-  onRowDragEnd: (e: RowDragEvent<Character>) => void,
-  onRowDragLeave: (e: RowDragEvent<Character>) => void,
-  navigateToNextCell: (x: NavigateToNextCellParams<Character>) => CellPosition | null,
-  isExternalFilterPresent: (x: IsExternalFilterPresentParams<Character>) => boolean,
-  doesExternalFilterPass: (x: IRowNode<Character>) => boolean,
-}) {
-  const { t } = useTranslation(['charactersTab', 'common', 'gameData'])
+const defaultColDef: ColDef<Character> = {
+  sortable: false,
+  cellStyle: { display: 'flex' },
+}
+
+const gridOptions: GridOptions<Character> = {
+  rowHeight: 46,
+  rowDragManaged: true,
+  animateRows: true,
+  suppressDragLeaveHidesColumns: true,
+  suppressScrollOnNewData: true,
+}
+
+export function CharacterGrid() {
+  const { t } = useTranslation('charactersTab', { keyPrefix: 'GridHeaders' })
+  const { t: tGameData } = useTranslation('gameData', { keyPrefix: 'Characters' })
+  const filters = useCharacterTabStore((s) => s.filters)
   const [characterRows, setCharacterRows] = useState(DB.getCharacters())
   window.setCharacterRows = setCharacterRows
```

Comment:
This should probably be swapped out for the useCharacterTabStore instead of maintaining its own local state